src/coreclr/src/jit/importer.cpp

```diff
@@ -20742,7 +20756,7 @@ bool Compiler::impCanSkipCovariantStoreCheck(GenTree* value, GenTree* array)
     }
 
     // Check for T[] with T exact.
-    if (!impIsClassExact(arrayElementHandle))
+    if (!impIsClassExact(arrayElementHandle, false))
     {
```

Comment:
if I enable `impIsClassExact` for arrays by default this call will cause ~400bytes regression. Example: https://www.diffchecker.com/nKWlHaG7

Reply:
That's a size regression but quite likely a perf improvement -- we are now bypassing the covariant store check and just need a write barrier.

The store check helper also does bounds checking, which is now explicit and is what causes the size growth.

Reply:
That's a size regression but quite likely a perf improvement -- we are now bypassing the covariant store check and just need a write barrier.

The store check helper also does bounds checking, which is now explicit and is what causes the size growth.

Reply:
That's a size regression but quite likely a perf improvement -- the code no longer needs the covariant store check, it just needs a write barrier.

The store check helper also does bounds checking, which now must be done in the jitted code, and is what causes the size growth.

Reply:
Github acting weird.

Reply:
ah, ok, I thought it introduced a new bound check. Will post the new diff later then.